### DISCUSSION TITLE
Extra Syntax Inserted by Completions -- How to Use or Suppress?

### DISCUSSION THREAD

**Discussion by ewhac** [neoclide/coc.nvim#4910] [Q&A]
I'm wondering where some extra "stuff" is coming from when I select completions.  As an example, let's say I want to call a method off a `table` struct.  I start by typing `table.`:
![image](https://github.com/neoclide/coc.nvim/assets/2722614/0b7ad59e-d61f-448a-a67e-e6cbdecf0eb0)

I narrow the search by further typing `add`.  I then move the selection to the entry for `add_row()`:
![image](https://github.com/neoclide/coc.nvim/assets/2722614/c5966f2a-71a5-48c2-a21d-472029a67b81)

And then I press return to select it:
![image](https://github.com/neoclide/coc.nvim/assets/2722614/2c2a2b80-9cd5-4e03-a178-e3b29ef36a78)

My question is concerning the `(${1:row})$0` part that got inserted.  Where does it come from?  What is it used for?  How would I make use of it or, failing that, how would I suppress it, leaving me with just the opening paren?

**Comment by fannheyward** [comment]
It's snippets from language server. Do you have disabled `suggest.snippetsSupport` in your coc-settings.json? By default, coc.nvim will apply the snippet, put your cursor to `${1:row}`, and can jump with `<C-j>` to `$0`.

**Reply by ewhac** [reply]
> Do you have disabled `suggest.snippetsSupport` in your coc-settings.json?

Indeed, I have `suggest.snippetsSupport` expressly set to `false`.  It seemed appropriate because I don't have a snippets provider installed.

**Reply by fannheyward** [reply]
Language server also provides snippets, coc.nvim can handle to expand snippets for this.

**Reply by ewhac** [reply]
Ah, didn't realize that.  So, until I can figure out how to fine-tune the inserted snippets, it looks like I have to disable snippet generation in each LSP I use.
